### PR TITLE
Win32: Drop extra leading underscore for D symbol mangling

### PR DESCRIPTION
**Pull request by kinke**
Compatible with DMD.

### CODE REVIEW

File: gen/mangling.cpp [hunk 1]
```diff
@@ -97,8 +97,8 @@ std::string hashSymbolName(llvm::StringRef name, Dsymbol *symb) {
 }
 }
 
-std::string getMangledName(FuncDeclaration *fdecl, LINK link) {
-  std::string mangledName(mangleExact(fdecl));
+std::string DtoMangledName(FuncDeclaration *fdecl, LINK link) {
```
Comment (JohanEngelen) [inline gen/mangling.cpp]: please don't change these function names to the awful "Dto*" pattern.

File: gen/moduleinfo.cpp [hunk 2]
```diff
@@ -56,11 +57,11 @@ llvm::Function *buildForwarderFunction(
   const auto fnTy =
       LLFunctionType::get(LLType::getVoidTy(gIR->context()), {}, false);
 
-  std::string const symbolName = gABI->mangleFunctionForLLVM(name, LINKd);
-  assert(gIR->module.getFunction(symbolName) == NULL);
+  const auto llMangle = DtoMangledFuncName(name, LINKd);
```
Comment (JohanEngelen) [inline gen/moduleinfo.cpp]: you are changing a lot of variable names, where the new name is worse than before. How is `llMangle` better than `symbolName` ? If you want to indicate that it is mangled, then change it to `mangledSymbolName`.

File: ir/irclass.cpp [hunk 3]
```diff
@@ -124,19 +130,15 @@ LLGlobalVariable *IrAggr::getInterfaceArraySymbol() {
   LLType *InterfaceTy = DtoType(Type::typeinfoclass->fields[3]->type->nextOf());
 
   // create Interface[N]
-  LLArrayType *array_type = llvm::ArrayType::get(InterfaceTy, n);
+  const auto llMangle = DtoMangledInterfaceInfosSymbolName(cd);
```
Comment (JohanEngelen) [inline ir/irclass.cpp]: `mangledName` --> `llMangle` ?

File: gen/moduleinfo.cpp (hunk 2 above)
Comment (kinke) [reply gen/moduleinfo.cpp]: All purely subjective. There are 3 mangles - base mangle, LL mangle, final mangle.

File: gen/mangling.cpp (hunk 1 above)
Comment (kinke) [reply gen/mangling.cpp]: I find it absolutely appropriate, as it translates from D layer to LLVM layer.

File: gen/moduleinfo.cpp (hunk 2 above)
Comment (JohanEngelen) [reply gen/moduleinfo.cpp]: Of course subjective. But you can do a lot better here.
All these abbreviations are bad for legibility. Just write down in the code what you are doing: e.g. `irMangledSymbolName`.

File: gen/mangling.cpp (hunk 1 above)
Comment (JohanEngelen) [reply gen/mangling.cpp]: `getMangledNameForIR`, `getIRMangledName`, something like that is much clearer than "D to".

Comment (kinke) [reply gen/mangling.cpp]: Renamed.

**Pull request merged**